Signup page

Haroonabdulrazaq opened the pull request:
#### What does this PR do?
It creates a clone for Mint.com Signup Page

#### Description of Task to be completed?
1. Create a header with a sign-in button
2.  Create a signup form
3. Create a page footer

#### How should this be manually tested?
git clone https://github.com/Haroonabdulrazaq/Mint.git
cd into the project folder

paulzay committed:
Create basic structure of the project

Add reset.css file to remove initial css stylings.

Haroonabdulrazaq committed:
Add form structure

paulzay committed:
Add styling for the signup button and the form input

paulzay committed:
add more styling to the form

paulzay committed:
add font-awesome icon to the signup button

paulzay committed:
Add recapture div and styling to form

Haroonabdulrazaq committed:
Style the form footer

paulzay committed:
Add styling for the footer

Add margin-top and link color.

paulzay committed:
fix validator issues

Haroonabdulrazaq committed:
Add stickler config files

Haroonabdulrazaq committed:
Merge branch 'signup-page' of https://github.com/Haroonabdulrazaq/Mint into signup-page

Haroonabdulrazaq committed:
Add styles to main footer

Haroonabdulrazaq committed:
Add main header and footer

paulzay committed:
add lock icons and header styling

paulzay committed:
add intuit logos

Haroonabdulrazaq committed:
Add styling to sub-logo

paulzay committed:
add intuit logo and styling

Haroonabdulrazaq committed:
style password input

Haroonabdulrazaq committed:
Add Readme.md file

paulzay committed:
fix css error, edit README and LICENCE

Haroonabdulrazaq committed:
Add screenshots

paulzay committed:
Merge branch 'gh-pages' of https://github.com/Haroonabdulrazaq/Mint into signup-page

paulzay committed:
Add live demo link

Haroonabdulrazaq committed:
Add screenshot

Haroonabdulrazaq committed:
Merge branch 'signup-page' of https://github.com/Haroonabdulrazaq/Mint into signup-page

davidmrtz-dev reviewed (changes requested):
## Hi Team. 👨‍🚀  
### Very good job so far.
However, the following issues need to be fixed before moving on  👨‍💻  ⬇️ 

****

- [ ] Stickler.

There's an error shown by the stickler lint, its related to stylelint.config.js. Please fix that.
****

- [ ] Form Section.

You did a great job building this part of the page but we can improve, please check the following points.

1.  You have too much space between the logos. Reduce that space following the given images.

2. The box containing the form is too large, please reduce the size of the box containing the form according to the page on which this project is based.

3. The section with the text 'Create an Intuit account' and 'One account for everything Intuit. Learn more' in the form looks different than that you have on your page, please use the same type of font of the given page also add the color blue to the link.

4. The input field sizes are too small according to the given page. Use the 95% of the width on each input field to follow the page that this project is based on.

5. The 'Create Account' button has the same width issue as the input fields also the blue color of the links of the terms and conditions section. Kindly fix that and add the required styling.

6. The captcha section has a different background color on the original page, also add the blue color to the links. 

Remember: We are building a clone of the page, so keep eyes on details 😄 😉 

Your page with references to the listed points above. ⬇️ 

![Screen Shot 2020-05-05 at 9 19 10 PM](https://user-images.githubusercontent.com/52430951/81133489-68977e80-8f17-11ea-8550-4989285c13f3.png)


The example on which the project is based. ⬇️ 

![Screen Shot 2020-05-05 at 9 23 48 PM](https://user-images.githubusercontent.com/52430951/81133498-6d5c3280-8f17-11ea-82aa-a5b0776973c9.png)


****

- [ ] Id's.

You have defined some Id's on your HTML that you never use on your CSS, please remove them and leave only those ids/classes that you are going to use.
The use of classes over ids is preferred in accordance with good coding practices.
Heres a helpful [article](https://speckyboy.com/good-bad-css-practices) that points out why is this important. 

****

### **Please kindly make the changes required and submit for re-review once you are ready!** 🙏 

****

### Status: Changes required. ♻️  👀 
****

Happy coding! 😄 💻  

**[TSE:  David Elí]** 👨‍💻 
Slack: @ David Eli Martinez Garcia

paulzay committed:
add color to links and space the logos

paulzay committed:
add color to the links and space the logos

paulzay committed:
Merge branch 'signup-page' of https://github.com/Haroonabdulrazaq/Mint into signup-page

Haroonabdulrazaq committed:
fix style as reviewed

paulzay committed:
add stylelint.config.js

paulzay committed:
fix tabs spacing on the css

paulzay committed:
reduce tab size to 2

paulzay committed:
fix linter error

paulzay committed:
add newline spacing

paulzay committed:
fix lint issue

Haroonabdulrazaq committed:
fix all error in the syles

Haroonabdulrazaq committed:
Add style to the header link

Haroonabdulrazaq committed:
Add id's to the form inputs

addod19 commented:
## Pull Request under review by Daniel Larbi Addo (TSE)

addod19 commented:
## Pull Request under review by Daniel Larbi Addo (TSE)

addod19 reviewed (changes requested):
## Status: Require changes :recycle: :traffic_light: 

Hi ***TEAM***, congratulations on fixing the comments pointed out by the previous TSE's :smile: However, we left one comment we need to address before we proceed. Kindly address the comment below:
- Kindly remove all ```ids``` that are not used in the code

[OPTIONAL]
Kindly apply the DRY rule so we avoid repeat code in our CSS file. This may [help](https://speckyboy.com/good-bad-css-practices/)

Kindly make this change and submit another code review :smile: 
_Happy coding_ :man_technologist: 
Daniel Larbi Addo on slack

Haroonabdulrazaq committed:
Add IDS in html and css doc

lucasmazo32 commented on assets/stylesheets/index.css:
```diff
@@ -0,0 +1,248 @@
+  border: 1px solid #c1c5c8;
+}
+
+#email {
+  background: white;
+}
+
+#number {
+  background: white;
+}
+
+#password {
+  background: white;
+}
```
Hello Team, although the IDs are used you could have used a class with the property ```background: white;```.

lucasmazo32 reviewed (changes requested):
#### Hello Team, there is one change you have to make before this project is approved.

The IDs implemented in your code can be easily changed by a class, and you wouldn't have the need to create more CSS lines of code.

Please change the IDs to a class and request another code review.

### Happy coding! :computer:

## Lucas Mazo

lucasmazo32 commented:
# Status: Changes required 🔧

Haroonabdulrazaq commented:
The labels requires an I'd for the HTML validator to pass, then the former
TSE requested no unused ID, I.e use all ID in the HTML
I'm confused you're requesting all ID should be changed to classes. What to
do?

On Friday, May 8, 2020, Lucas Mazo <notifications@github.com> wrote:

> *@lucasmazo32* requested changes on this pull request.
> Hello Team, there is one change you have to make before this project is
> approved.
>
> The IDs implemented in your code can be easily changed by a class, and you
> wouldn't have the need to create more CSS lines of code.
>
> Please change the IDs to a class and request another code review.
> Happy coding! 💻 Lucas Mazo
> ------------------------------
>
> In assets/stylesheets/index.css
> <https://github.com/Haroonabdulrazaq/Mint/pull/1#discussion_r421864295>:
>
> > +#email {
>
> +  background: white;
>
> +}
>
> +
>
> +#number {
>
> +  background: white;
>
> +}
>
> +
>
> +#password {
>
> +  background: white;
>
> +}
>
>
> Hello Team, although the IDs are used you could have used a class with the
> property background: white;.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Haroonabdulrazaq/Mint/pull/1#pullrequestreview-407917898>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMDDA2ZS4NC77T2B3D6HLATRQNFAJANCNFSM4MZVIY3A>
> .
>

Haroonabdulrazaq committed:
Changed IDs to classes

lucasmazo32 commented:
> 
> The labels requires an I'd for the HTML validator to pass, then the former TSE requested no unused ID, I.e use all ID in the HTML I'm confused you're requesting all ID should be changed to classes. What to do?
> […](#)
> On Friday, May 8, 2020, Lucas Mazo ***@***.***> wrote: ***@***.**** requested changes on this pull request. Hello Team, there is one change you have to make before this project is approved. The IDs implemented in your code can be easily changed by a class, and you wouldn't have the need to create more CSS lines of code. Please change the IDs to a class and request another code review. Happy coding! 💻 Lucas Mazo ------------------------------ In assets/stylesheets/index.css <[#1 (comment)](https://github.com/Haroonabdulrazaq/Mint/pull/1#discussion_r421864295)>: > +#email { + background: white; +} + +#number { + background: white; +} + +#password { + background: white; +} Hello Team, although the IDs are used you could have used a class with the property background: white;. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub <[#1 (review)](https://github.com/Haroonabdulrazaq/Mint/pull/1#pullrequestreview-407917898)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AMDDA2ZS4NC77T2B3D6HLATRQNFAJANCNFSM4MZVIY3A> .

You are right. IDs that are used for labels don't necessarily have to be in the CSS file.

Kindly submit for another PR and your project should get approved.

## Lucas Mazo

Haroonabdulrazaq committed:
Change all ID to classes

sumancrest0001 commented:
**Status: Changes required
Nice work Team :clap: :clap:  :+1:** 

You have ticked most of the requirements from the Microverse curriculum but we need to fix a couple of issues before moving to the next project. Kindly resolve the following issues.
[Required] It would be nice if you align logos and texts at the center vertically.
It should be like this :point_down: 
![Screenshot from 2020-05-09 13-09-23](https://user-images.githubusercontent.com/37747619/81467165-9a9a3080-91f6-11ea-92eb-f77d3e20f943.png)

currently, in your project :point_down: 
![Screenshot from 2020-05-09 13-12-00](https://user-images.githubusercontent.com/37747619/81467182-ba315900-91f6-11ea-983b-167f8366ea89.png)

sumancrest0001 commented on assets/stylesheets/index.css:
```diff
@@ -0,0 +1,248 @@
+  margin-left: 10px;
+}
+
+.signup img {
+  display: inline-block;
+  width: 16px;
+  height: 12px;
+  margin-top: 2px;
+  margin-right: 4px;
+  vertical-align: text-top;
+}
+
+.mini-header {
+  margin-top: 20px;
+}
+
+.header img {
+  display: inline-block;
+  width: 16px;
+  height: 12px;
+  margin-top: 2px;
+  margin-right: 4px;
+  vertical-align: text-top;
+}
+
```
[Required] We have to avoid the repetition of code to make our project professional. Follow the DRY principle. [Pathright requirement](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/html_css/section/css_bad_practices_requirement.png). Reference this [article](https://speckyboy.com/good-bad-css-practices)

sumancrest0001 commented on assets/stylesheets/index.css:
```diff
@@ -0,0 +1,248 @@
+  border: 1px solid #c1c5c8;
+}
+
+.email {
+  background: white;
+}
+
+.number {
+  background: white;
+}
+
+.password {
+  background: white;
+}
+
```
[Required] Remove all the repeated styling. you can do something like this :point_down: 
`.email, .number, .password {
  background: white;
}`

sumancrest0001 reviewed (changes requested):
After fixing the above-requested issues, send for another review. I am confident that it won't take too long to finish.
Need help or have questions? Feel free to contact me on Slack **(Suman Shrestha)**

**Happy coding !!!!**

paulzay committed:
remove repeated styling

paulzay committed:
Implement the review feedback

1. Center the sublogo images and texts.
2. Modify the signup button.
3. Remove duplication in the css code

paulzay committed:
fix linting issues

lucasmazo32 reviewed (approved):
# Project approved :rocket:

#### Great job!

#### Your project looks great!

### You can go ahead and merge this pull request!

### Happy coding! :computer:

## Lucas Mazo
